android-core/plugins/org.eclipse.andmore/src/org/eclipse/andmore/internal/build/builders/BaseBuilder.java

```diff
@@ -320,10 +320,16 @@ protected void abortOnBadSetup(@NonNull IJavaProject javaProject,
         if (projectState == null) {
             projectState = Sdk.getProjectState(javaProject.getProject());
         }
-
+        
+        // the project state can still be null, 
+        // e.g. if there is no project.properties file
+        if (projectState == null) {
```

Comment:
I understand the need for unit tests and I would be willing to provide them, but  the integration test plugin is not compliling on my machine from within Eclipse and it will take more time than I currently have to find out why this is the case. 

The fixes I have provided are working on my Mac and my Windows machine and some of the associated bugs make the plugin completely unusable for Android development. For example, the broken language selection algorithm breaks the visual ui editor and not being able to pass aapt a no-vector-versions argument prevents me and anyone else that trys to use a newer version of the support library (that's probably everyone that is not just implementing an app for Android 6 or 7) from compiling an app that will also work on Android 4.

Anyway, if I have some slack time again, I will try to look at the problem with the test plugin. 

PS: I think you should consider changing your strategy for motivating "people to become committers". I know of at least one example where it is about to fail.
PPS: Thanks for spending your time to keep the Android Eclipse Plugin alive so my brain does not have to operate at jet speed.